### ISSUE TITLE
Few bugs with 'Freeze/unfreeze' shortcuts

### ISSUE THREAD

**Issue by resiehnnes** [MuntashirAkon/AppManager#1040]
### Please check before submitting an issue

- [X] I know what my device, OS and App Manager versions are
- [X] I know how to take logs
- [X] I know how to reproduce the issue which may not be specific to my device

### Describe the bug

First problem:

After creating a shortcut for already **freezed** app (long press 'Freeze/unfreeze' button) and setting up 'Open app after unfreezing' and 'Automatically freeze app when the phone is locked' the shortcut is placed on home screen but the icon is darker then it supposed to be, and gets completely black when the app is freezed after locking and unlocking phone (when 'Automatically freeze app when the phone is locked' is triggered)

Second problem:

'Automatically freeze app when the phone is locked' trigger works only for the first time and fails to freeze app after first 'lock/unlock' cycle leaving 'Waiting for the phone to be locked...' active despite further locking and unlocking the phone.

### Device info

- Device: Huawei P30 Pro (VOG-L29)
- OS Version: Android 10
- App Manager Version: 3.1.0 Stable
- Mode: ADB, no root present

**Comment by MuntashirAkon** [comment]
> After creating a shortcut for already **freezed** app (long press 'Freeze/unfreeze' button) and setting up 'Open app after unfreezing' and 'Automatically freeze app when the phone is locked' the shortcut is placed on home screen but the icon is darker then it supposed to be, and gets completely black when the app is freezed after locking and unlocking phone (when 'Automatically freeze app when the phone is locked' is triggered)

I'm not sure what's causing it. Did you check that the launcher is not improvising anything? Some launchers seem to apply certain adaptive filters which can cause this.

> 'Automatically freeze app when the phone is locked' trigger works only for the first time and fails to freeze app after first 'lock/unlock' cycle leaving 'Waiting for the phone to be locked...' active despite further locking and unlocking the phone.

I don't really know if this issue can be fixed. Detecting phone lock isn't an easy thing to do since Android does not provide an API to monitor that. But I'll see if I can reproduce it.

**Comment by resiehnnes** [comment]
> I'm not sure what's causing it. Did you check that the launcher is not improvising anything? Some launchers seem to apply certain adaptive filters which can cause this.

I am using stock launcher with stock icons if that makes any difference. I also tried with 3rd party launcher which has same problem. Made some screenshots so you can see how it looks.

App info:

![01](https://user-images.githubusercontent.com/52158526/229366246-b974c785-a1a8-4fa3-8ee3-976fa71d9013.jpg)

Creating a shortcut for this app:

![03](https://user-images.githubusercontent.com/52158526/229366341-338c41d1-f7b1-44bc-8e5c-4d40f8b12afa.jpg)

Shortcut is placed on homescreen:

![04](https://user-images.githubusercontent.com/52158526/229366408-5701afa0-131e-4dcb-a03b-0fedec8dfc6a.jpg)

Shortcut after 'Freeze/unfreeze' cycle:

![05](https://user-images.githubusercontent.com/52158526/229366445-c01f1076-4a05-4a72-a154-7a4192bcea6e.jpg)

And it stays black if I keep freezing or unfreezing app, but gets back to normal if I do close AppManager from recent tasks and use the shortcut again, but then shortly it becomes darker, and completely black.

> I don't really know if this issue can be fixed. Detecting phone lock isn't an easy thing to do since Android does not provide an API to monitor that. But I'll see if I can reproduce it.

Yesterday it seemed that AppManager were able to detect first phone lock successfully and freezed the used app, but today for some reason 'Automatically freeze app when the phone is locked' doesn't get the job done even after the first 'lock/unlock' cycle.

**Comment by MuntashirAkon** [comment]
> I am using stock launcher with stock icons if that makes any difference. I also tried with 3rd party launcher which has same problem. Made some screenshots so you can see how it looks.

It's weird. Since the app was already frozen, the shortcut offered was a dimmed version of the icon. So, when you click on the icon again, i.e. unfreeze the app, it should return to the original icon. I have to see if there's an issue with updating shortcuts in Android 10.

**Comment by resiehnnes** [comment]
@MuntashirAkon Yep, when frozen app with dimmed shortcut is launched the icon stays dimmed, and gets black when I freeze app using the same shortcut.

**Comment by MuntashirAkon** [comment]
Tested on three devices regarding the dimming of icons:

1. Android 11 - Lineage OS 18.1 (my device, working as expected)
2. Android 13 - Emulator (working as expected) [^a13]
3. Android 8.1 - Emulator (working as expected) [^a8.1]

So, this might be emitted by something else e.g. adaptive icons or other icon packs when enabled.

[^a13]: [A13-Freeze-Test.webm](https://user-images.githubusercontent.com/8176128/236697295-ab3be9eb-f811-46f8-8bae-6bb8bfbeb24b.webm)
[^a8.1]: [A8.1-Freeze-Test.webm](https://user-images.githubusercontent.com/8176128/236697340-51f01704-1496-4941-b85c-a93a1fa4f002.webm)

**Comment by MuntashirAkon** [comment]
It looks like icons are not dimmed at all in the production releases, but works in debug releases. Weird.

**Comment by shuvashish76** [comment]
> 'Automatically freeze app when the phone is locked' trigger works only for the first time...

For me it doesn't work at all. Android 12, mode: root, latest debug version.

> Detecting phone lock isn't an easy thing to do since Android does not provide an API to monitor that.

Request to check how "block all apps on device locked" firewall feature works for RDNS. (ref: [link](https://github.com/celzero/rethink-app/issues/29))

**Comment by MuntashirAkon** [comment]
Here's the definition of what App Manager means by locked phone:

> The phone is secured by a PIN, pattern or password, and the phone or a SIM card is currently locked.

If you do not use these, the phone will not be considered locked, and the apps won't be frozen.
